code/script.js

```diff
@@ -230,96 +323,176 @@ const setSecret = () => {
   const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  if (Array.isArray(secret[category])) {
     if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
     } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
-    }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    };
+  } 
+   else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
-    }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    };
+  };
+  generateBoard()
+};
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
+  let personToCheck = personToConfirm
   // store the interaction from the player in a variable.
   // remember the confirm() ?
+  let userConfirmed = window.confirm(`Do you want to guess ${personToConfirm}?`)
   // If the player wants to guess, invoke the checkMyGuess function.
-}
+  if (userConfirmed) {
+    checkMyGuess(personToCheck)
+  }
+  else {
+    alert("Guess cancelled")
+  };
+};
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
+  const userHasGuessedCorrectly = (secret.name === personToCheck);
+  displaySecret.src = secret.img
+  secretText.innerHTML = `The mushroom you were identifing was ${secret.name}. Its Swedish name is ${secret.swedishName} and its Latin name is <em>${secret.latinName}</em>. It's easy to find mushrooms in the Swedish forests, but when you identify one you must be sure to look at all its different characteristics, as there are many lookalikes. Don't eat anything if you're not sure!`
```

Comment:
So cool that you are repeating all the properties of the object in the secret text!